service/cziti/mfa.go

```diff
@@ -157,6 +138,53 @@ func ReturnMfaCodes(id *ZIdentity, code string) ([]string, error) {
 	}
 }
 
+// used when an identity is partially authenticated and waiting on 2fa, i.e. for authentication and for timeouts
+func AuthMFA(id *ZIdentity, code string) error {
+	ccode := C.CString(code)
+	defer C.free(unsafe.Pointer(ccode))
+
+	log.Tracef("authenticating MFA for fingerprint: %s using code: %s", id.Fingerprint, code)
+	C.ziti_mfa_auth(id.czctx, ccode, C.ziti_mfa_cb(C.ziti_auth_mfa_status_cb_go), unsafe.Pointer(C.CString(id.Fingerprint)))
+	authResult := strings.TrimSpace(<-mfaAuthResults)
```

Comment:
was this trim actually needed?

Reply:
This auth function was already there in the code. I just changed the c function call in this Auth function.